src/Uno.UI/NativeFramePresenter.Android.cs

```diff
@@ -145,10 +144,32 @@ private async Task UpdateStack(PageStackEntry entry, NavigationEventArgs e)
 						await oldPage.AnimateAsync(GetExitAnimation());
 						oldPage.ClearAnimation();
 					}
+
 					if (oldPage != null)
 					{
 						_pageStack.Children.Remove(oldPage);
 					}
+
+					if (!FeatureConfiguration.NativeFramePresenter.AndroidUnloadInactivePages)
+					{
+						for (var pageIndex = _pageStack.Children.Count-1; pageIndex >= 0; pageIndex--)
```

Comment:
```suggestion
 						// Remove pages from the grid that may have been removed from the BackStack list
 						// Those items are not removed on BackStack list changes to avoid interfering with the `GoBack` method's behavior.
						for (var pageIndex = _pageStack.Children.Count-1; pageIndex >= 0; pageIndex--)
```